pyomo/gdp/plugins/chull.py

```diff
@@ -840,5 +853,131 @@ def _xform_constraint(self, obj, disjunct, infodict, var_substitute_map,
+                           transformation as a disaggregated variable 
+                           (and so appears on a transformation block 
+                           of some Disjunct)
+        """
+        transBlock = disaggregated_var.parent_block()
+        try:
+            src_disjunct = transBlock._srcDisjunct()
+        except:
+            raise GDP_Error("%s does not appear to be a disaggregated variable"
+                            % disaggregated_var.name)
+        return transBlock._disaggregatedVarMap['srcVar'][disaggregated_var]
+
+    # retrieves the disaggregation constraint for original_var resulting from
+    # transforming disjunction
+    def get_disaggregation_constraint(self, original_var, disjunction):
+        """
+        Returns the disaggregation (re-aggregation?) constraint 
+        (which links the disaggregated variables to their original) 
+        corresponding to original_var and the transformation of disjunction.
+
+        Parameters
+        ----------
+        original_var: a Var which was disaggregated in the transformation
+                      of Disjunction disjunction
+        disjunction: a transformed Disjunction containing original_var
+        """
+        for disjunct in disjunction.disjuncts:
+            transBlock = disjunct._transformation_block
+            if not transBlock is None:
+                break
+        if transBlock is None:
+            raise GDP_Error("Disjunction %s has not been properly transformed: "
+                            "None of its disjuncts are transformed." 
+                            % disjunction.name)
+        try:
+            return transBlock().parent_block()._disaggregationConstraintMap[
+                original_var][disjunction]
+        except:
+            raise GDP_Error("It doesn't appear that %s is a variable that was "
+                            "disaggregated by Disjunction %s" % 
+                            (original_var.name, disjunction.name))
+
+    def get_var_bounds_constraint(self, v):
+        """
+        Returns the IndexedConstraint which sets a disaggregated
+        variable to be within its bounds when its Disjunct is active and to
+        be 0 otherwise. (It is always an IndexedConstraint because each
+        bound becomes a separate constraint.)
+        
+        Parameters
+        ----------
+        v: a Var which was created by the chull  transformation as a 
+           disaggregated variable (and so appears on a transformation 
+           block of some Disjunct)
+        """
+        # This can only go well if v is a disaggregated var
+        transBlock = v.parent_block()
+        try:
+            return transBlock._bigMConstraintMap[v]
+        except:
+            raise GDP_Error("Either %s is not a disaggregated variable, or "
+                            "the disjunction that disaggregates it has not "
+                            "been properly transformed." % v.name)
```

Comment:
This is dangerous as it can hide underlying bugs.  Either catch a specific exception, or (probably better) log this message as an error and re-raise the original exeception.